### PR TITLE
feat: added Tiplink feature

### PR DESCRIPTION
**Pull request by shivaji43**
Added a Tiplink function where agent can create a tiplink for a given SOL amount  #21

### CODE REVIEW

**Comment by arihantbansal** [comment]
Please create a langchain tool for Tiplink function in https://github.com/sendaifun/solana-agent-kit/blob/main/src/langchain/index.ts and export it via `createSolanaTools`

**Comment by shivaji43** [comment]
@arihantbansal Please take a look if everything is correct

File: src/langchain/index.ts [hunk 1]
```diff
@@ -7,6 +7,7 @@ import { fetchPrice } from "../tools/fetch_price";
 import { BN } from "@coral-xyz/anchor";
 import { FEE_TIERS } from "../tools";
 import { toJSON } from "../utils/toJSON";
+import { TipLink } from "@tiplink/api"
```
Comment (arihantbansal) [inline src/langchain/index.ts:10]: why is this import needed?

**Review by arihantbansal — CHANGES_REQUESTED** [review]

**Comment by thearyanag** [comment]
hey @shivaji43 - can you modify the code to also support SPL tokens?

**Comment by shivaji43** [comment]
> hey @shivaji43 - can you modify the code to also support SPL tokens? 

Yes I will do that

**Comment by shivaji43** [comment]
@thearyanag please take a look

**Comment by thearyanag** [comment]
can you pls rebase the branch

**Comment by thearyanag** [comment]
@shivaji43 has this code being tested?

File: src/langchain/index.ts [hunk 2]
```diff
@@ -980,6 +980,45 @@ export class SolanaOpenbookCreateMarket extends Tool {
...
+
+  protected async _call(input: string): Promise<string> {
+    try {
+      const amountSol : number = parseFloat(input);
+
+      if (isNaN(amountSol) || amountSol <= 0) {
+        throw new Error("Invalid amount. Please provide a valid amount of SOL.");
+      }
+
+      const { url, signature } = await this.solanaKit.createTiplink(amountSol);
```
Comment (thearyanag) [inline src/langchain/index.ts:1004]: splMintAddress is part of the func, but not getting passed even :/ 

pls test the code before raising a PR

Comment (shivaji43) [reply src/langchain/index.ts:1004]: I'm sorry , I will do the changes today test the code and then ask for review

Comment (shivaji43) [reply src/langchain/index.ts:1012]: I will try to make a new pull request , I have tested the code and it is working

**Pull request closed without merging**